x-pack/legacy/plugins/apm/server/lib/traces/get_trace_root.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchParams } from 'elasticsearch';
+import { idx } from '@kbn/elastic-idx';
+import {
+  PROCESSOR_EVENT,
+  TRACE_ID,
+  PARENT_ID
+} from '../../../common/elasticsearch_fieldnames';
+import { Transaction } from '../../../typings/es_schemas/ui/Transaction';
+import { rangeFilter } from '../helpers/range_filter';
+import { Setup } from '../helpers/setup_request';
+
+export async function getTraceRoot(traceId: string, setup: Setup) {
+  const { start, end, client, config } = setup;
+
+  const params: SearchParams = {
+    index: [
+      config.get('apm_oss.spanIndices'),
+      config.get('apm_oss.transactionIndices')
+    ],
+    body: {
+      size: 1,
+      query: {
+        bool: {
+          filter: [
+            { term: { [TRACE_ID]: traceId } },
+            { terms: { [PROCESSOR_EVENT]: ['transaction'] } },
+            { range: rangeFilter(start, end) }
+          ],
+          must_not: {
+            exists: { field: PARENT_ID }
+          }
+        }
+      }
+    }
+  };
+
+  const resp = await client.search<Transaction>(params);
+  const root: Transaction | undefined = idx(resp, _ => _.hits.hits[0]._source);
+
+  return root;
+}
```

Comment:
Instead of fetching the root transaction separately, I think we can add a `should` clause to the existing `getTraceItems` so that the root transaction will have higher score, and thus will be the very first item in the list:

```js
          should: {
            exists: { field: PARENT_ID }
          }
```

Additionally we should sort by `duration` to get the longest transactions/spans since they are often in the beginning:
```js
"sort": [
        { "_score": { "order": "desc" }}
        { "transaction.duration.us":   { "order": "desc" }}, // this could be a little tricky since span docs don't have this field
        { "span.duration.us":   { "order": "desc" }} // this could be a little tricky since transaction docs don't have this field
    ]
```

And finally I think we should also show a warning message if there are items that are not being shown.

This will not solve the problem with max 1000 items but it'll hopefully make it a bit better until we can make the "recursive loading" solution we've talked about

Reply:
I can see that you've added some commits to this PR. It's still the intention to get rid of this call (`getTraceRoot`), right?

Reply:
After some initial difficulty, @dgieselaar and I found right right combination of `should` clause and `sort` order to always return the root transaction in the same query in order to remove `getTraceRoot`.